### ISSUE TITLE
Make node changes work with edit transactions

### ISSUE THREAD

**Issue by matthew-carroll** [superlistapp/rich_text_editor#67]
Currently, nodes are changed directly by user interactions, e.g., indenting a list item.

All edits need to go through a transaction system for undo/redo behavior.

Find an appropriate solution for individual node changes.

**Comment by chenhaoaixuexi** [comment]
It's October 2023 now. why hasn't it been changed? how can I support undo redo? 😢

**Comment by WillianSalceda** [comment]
Any updates?

**Comment by matthew-carroll** [comment]
We've migrated to atomic commands, which was a critical pre-requisite for undo/redo. We're still vetting that system, which is comprised of requests, commands, and reactions.

We haven't started working directly on undo/redo yet.

**Comment by matthew-carroll** [comment]
This is being developed right now

**Comment by szpaqq** [comment]
Hey @matthew-carroll any news on progress? ;) 
Looking forward to test it out! :D

**Comment by simonbengtsson** [comment]
This would be awesome! My use case for this is native fidelity. Undo/redo with `cmd + z` and `cmd + shift + z` is the only thing I'm personally experiencing as being missing in super editor compared to Flutter's text field or native text fields on macOS such as those in Safari or Apple Notes etc (perhaps in addition to #2042  and #2075 on mobile). When writing normally I'm often using undo/redo to undo/redo a deletion or undo/redo pasted content etc.

The editor I'm working on is light and have no immediate requirements for custom components etc. The main reason I want to use a third party editor such as super editor compared to Flutter's text field is the the support for ordered and unordered lists in addition to the easier way to add support for styling such as bold and code formatted text.

**Comment by matthew-carroll** [comment]
@simonbengtsson can you clarify whether you're asking for this in `SuperEditor` or `SuperTextField`? One is a document editor, the other is a text field.

**Comment by simonbengtsson** [comment]
Right! I should have posted this in #515 since my use case is actually a text field I believe. It is applicable to both however ie the undo/redo use case as described above is the same for both document editors such as the one in Google Docs and text fields such as the one here on Github where I'm writing this comment.

**Comment by hannutho** [comment]
Hey! How is development going on this? When is a initial version expected to be released?

**Comment by matthew-carroll** [comment]
@hannutho please see https://github.com/superlistapp/super_editor/pull/1900 for the status of the work

**Comment by clotodex** [comment]
What is the status of this? Is this seen as complete? The only follow up issue i found is #2154
